### PR TITLE
New Azure Firewall Workbook with support for Structured Logs

### PR DESCRIPTION
**Pull request by gumoden**
## PR Checklist

* [ ] Explain your changes, so people looking at the PR know *what* and *why*, the code changes are the *how*.
* [ ] Validate your changes using one or more of the [testing](../Documentation/Testing.md) methods.

### If adding or updating templates:
* [ ] post a screenshot of templates and/or gallery changes
* [ ] ensure your template has a corresponding gallery entry in the gallery folder
* [ ] If you are adding a new template, add your team and template/gallery file(s) to the CODEOWNERS file. CODEOWNERS entries should be teams, not individuals
* [ ] ensure all steps have meaningful names
* [ ] ensure all parameters and grid columns have display names set so they can be localized
* [ ] ensure that parameters id values are unique __or they will fail PR validation__ (parameter ids are used for localization)
* [ ] ensure that steps names are unique __or they will fail PR validation__ (step names are used for localization)
* [ ] grep `/subscription/` and ensure that your parameters don't have any hardcoded resourceIds __or they will fail PR validation__
* [ ] remove `fallbackResourceIds` and `fromTemplateId` fields from your template workbook __or they will fail PR validation__

### CODE REVIEW

File: gallery/workbook/microsoft.network-azurefirewalls.json
Comment (gardnerjr) [inline gallery/workbook/microsoft.network-azurefirewalls.json:22]: this has the exact same name as the template above it?  the build should be failing that too if it isn't already.  names need to be unique.

Comment (gardnerjr) [reply gallery/workbook/microsoft.network-azurefirewalls.json:22]: and it is unclear to my why the names are so long?
the name of the group/category in the gallery is already "Network Insights AzureFirewalls" which isn't super useful, and then the items inside all start with "Network Insights AzureFirewalls" again?

these are folders/names that users see, why not just

"Network Insights" as the category and "Firewall Metrics" and "Structured Logs", etc. user friendly names.

Comment (gumoden) [reply gallery/workbook/microsoft.network-azurefirewalls.json:22]: I just made the changes on the folder and file names. I also fixed the single name of the template.

File: gallery/workbook/microsoft.network-azurefirewalls.json
Comment (gardnerjr) [inline gallery/workbook/microsoft.network-azurefirewalls.json:10]: i wouldn't move/rename the ids of workbooks, thse are ids, they might be in places opening them in code, depending on these in links/pins, etc.

my only comment as on the NAMES displayed to users.

File: Workbooks/Network Insights/AzureFirewallWorkbooks/StructuredLogsWorkbook/StructuredLogs.workbook
Comment (gardnerjr) [inline Workbooks/Network Insights/AzureFirewallWorkbooks/StructuredLogsWorkbook/StructuredLogs.workbook:14]: you keep on using the ARM Template version of this file.
in this repo, it should just be the workbook content.
when exporting from the workbook editor, do NOT switch it to ARM template mode, leave it in "gallery template" mode.
@ericajlin  i'm surprised this .workbook file being an arm template doesn't fail template validation or loc validation, etc

Comment (gumoden) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/StructuredLogsWorkbook/StructuredLogs.workbook:14]: Updated

File: gallery/workbook/microsoft.network-azurefirewalls.json
Comment (gumoden) [reply gallery/workbook/microsoft.network-azurefirewalls.json]: Updated

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gardnerjr) [inline Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook:7]: nit: is this header adding any information?  people know they are in workbooks already, and the title of the blade would already have the template name, so not sure this adds anything, but does take up important real eastate at the top of the window

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gardnerjr) [inline Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook:30]: still not screenshots to verify but this seems like a really long title for tabs.

"Azure Firewall - Application Rule log statistics" is the the title of a tab?

does it need azure firewall at all?

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gumoden) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook:7]: It is just a title for the workbook

image

I will bring this to the discussion with my team, but I would like to proceed without changing the title. Is that ok?
Soon I will probably be removing the previous 2 workbooks from the gallery, keeping only the new one.

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gumoden) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook:30]: This is how the tabs look like today:

image

I will discuss the tab names change with my team, but I don't want to have a dependency on this to publish the new workbook, since the other 2 workbooks are like that already. Is that ok?

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gumoden) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook:145]: It allows customers to query the data from different workspaces at once. So, they are able to have a central view of their Firewalls by using the same workbook. Is there a best practice to follow which meet this requirement?

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gardnerjr) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook:30]: unclear what that means.  why publish something subpar because something else is subpar?  i don't understand why the word "Azure Firewall" needs to appear 5 times at the top of this template.  on narrow monitors, all those tabs are going to end up in the "..." overflow (some already are?) and users won't even notice they are there.

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gardnerjr) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook:7]: that's true, but look:
image

the word azure firewall appears 10x already on the screen?

Comment (gumoden) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook:7]: Agreed on removing multiple references to Azure Firewall. Updated

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gumoden) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook]: Removed multiple references to Azure Firewall from the workbook gallery file

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gumoden) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook:145]: Removed value.

image

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gumoden) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook]: image

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gumoden) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook:7]: image

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gumoden) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook]: Hey @gardnerjr, do you have any additional comments to share? Is the PR good to be approved?

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gardnerjr) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook:7]: @gumoden i've only really been looking at it for "technical correctness / process", someone from network insights team should really be approving for content/behavior

File: Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook
Comment (gardnerjr) [reply Workbooks/Network Insights/AzureFirewallWorkbooks/NetworkInsights-AzureFirewallMetrics-StructuredLogs/NetworkInsights-AzureFirewallMetricsStructuredLogs.workbook]: @gumoden i've only been reviewing for technical things, where you were using the wrong formats of files, or general suggestions.

the network insights team is really the codeowners of this area, someone from that team really should be approving since they'd be generally responsible for it.

**Comment by gardnerjr** [comment]
@gumoden you need to have someone in https://github.com/orgs/microsoft/teams/network-insights approve this if you need to get it completed

**Comment by gumoden** [comment]
> @gumoden you need to have someone in https://github.com/orgs/microsoft/teams/network-insights approve this if you need to get it completed

Hi @gardnerjr, I've talked to them and @snagpal99 is reviewing the PR. Thank you!

**Review by snagpal99 — APPROVED** [review]

**Pull request merged**
